### PR TITLE
:sparkles: Add `stop_detached`

### PR DESCRIPTION
**Pull request by elbeno**
`stop_detached<Tag>()` may request cancellation of a sender started with `start_detached<Tag>(s)`.

### CODE REVIEW

File: docs/sender_consumers.adoc [hunk 1]
```diff
@@ -54,6 +64,33 @@ required.
 auto result = async::start_detached_unstoppable<struct Name>(s);
 ----
 
+=== `stop_detached`
+
+Found in the header: `async/start_detached.hpp`
+
+A sender started with `start_detached` may be cancelled with `stop_detached`,
```
Comment (mjcaisse-intel) [inline docs/sender_consumers.adoc:71]: todo: add docs for `Uniq` template parameter in `start_detached` section. The "same template argument" is not described there and current defaults to a unique lambda type.

**Review by mjcaisse-intel — APPROVED** [review]
Minor todo in docs. Looks great!

File: docs/sender_consumers.adoc (hunk 1 above)
Comment (elbeno) [reply docs/sender_consumers.adoc:71]: It is already documented there - was there something you wanted to add specifically?

Comment (mjcaisse-intel) [reply docs/sender_consumers.adoc:71]: Hmm... I missed it when looking at the markdown. Sorry.

I see that it is called out in the docs for being able to "hooking into the static allocation strategy". Maybe specify the template argument provides a "name" that can be used for providing a static allocation strategy and/or used in `stop_detached` to specify which sender to stop. If a name is given to a `start_detatched` and no user specified allocation strategy has been provided, the default strategy will be used.

Comment (elbeno) [reply docs/sender_consumers.adoc:76]: OK, I've added some explanation about that.

**Pull request merged**